### PR TITLE
Add monitoring to hyp3

### PR DESCRIPTION
**Pull request by Jlrine2**

### CODE REVIEW

File: apps/monitoring-cf.yml [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    Properties:
+      AlarmActions:
+        - !Ref AlarmTopic
+      AlarmDescription: Hyp3 processing failures
+      ComparisonOperator: GreaterThanThreshold
+      Period: 60
+      EvaluationPeriods: 1
+      Threshold: 0
+      Dimensions:
+        - Name: Processing Failures
```
Comment (asjohnston-asf) [inline apps/monitoring-cf.yml:26]: `Name` is the AWS-defined dimension to filter by, I believe `StateMachineArn` in this case

File: apps/monitoring-cf.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    Properties:
+      AlarmActions:
+        - !Ref AlarmTopic
+      AlarmDescription: Hyp3 api HTTP 500 errors
+      ComparisonOperator: GreaterThanThreshold
+      Period: 60
+      EvaluationPeriods: 1
+      Threshold: 0
+      Dimensions:
+        - Name: API 500 errors
```
Comment (asjohnston-asf) [inline apps/monitoring-cf.yml:45]: `ApiId` I think

File: apps/monitoring-cf.yml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+          Value: !Ref ApiId
+      MetricName: 5xxError
```
Comment (asjohnston-asf) [inline apps/monitoring-cf.yml:47]: just `5xx` I believe

**Comment by asjohnston-asf** [comment]
I think you can pick out the namespace/metric/dimension values by building the metric manually in Cloudwatch, then looking at `Details`, I think it's `<namespace> * <metric name> * <dimension name>: <dimension value>`

Screenshot from 2020-09-30 17-28-53

**Comment by asjohnston-asf** [comment]
We'll need to grant `cloudformation-deployment-role` additional permissions so it can manage alarms and topics, `sns:*` and `cloudwatch:*` would be sufficient

File: apps/monitoring-cf.yml (hunk 3 above)
Comment (Jlrine2) [reply apps/monitoring-cf.yml:47]: https://docs.aws.amazon.com/apigateway/latest/developerguide/api-gateway-metrics-and-dimensions.html
I think the console displays a simplified version

Comment (Jlrine2) [reply apps/monitoring-cf.yml:47]: although this documention reveals that our demention isn't valid

**Comment by Jlrine2** [comment]
APi alarm works now

**Comment by Jlrine2** [comment]
updated cloudformation-deployment-role

**Review by asjohnston-asf — APPROVED** [review]

**Pull request merged**